tiledb/ml/data_apis/pytorch_sparse.py

```diff
@@ -31,6 +32,34 @@ def __init__(self, x_array: tiledb.Array, y_array: tiledb.Array, batch_size: int
         self.y = y_array
         self.batch_size = batch_size
 
+    def __check_row_dims(self, x_row_idx: np.array, y_row_idx: np.array):
+        """
+        Check the row dimensionality of x,y in case y is sparse or not
+
+        Parameters:
+            x_row_idx (list): Expects the row indexes x_coords of x Sparse Array of the
+            dimension that is being batched
+
+            y_row_idx (list): if y Sparse Array -> Expects the row indexes y_coords of the
+            dimension that is being batched else if y is Dense Array -> data of y
+
+        Raises:
+            ValueError: If unique coords idx of x and y mismatch (both-sparse) or
+            when unique coords idx of x mismatch y elements when y is Dense
+        """
+        if isinstance(self.y, tiledb.SparseArray):
+            if set(x_row_idx) != set(y_row_idx):
```

Comment:
- Why `set()` here and `np.unique()` in the `else` clause?
- The exception message mentions "same number of rows" but you compare for same members; which one is it?

Reply:
Mainly is due to what I mention in the description of the ticket. In sparse representation the indexes and values that a slice returns are the non-zero ones. These values can be more than one per row, thus the indices can mainly have duplicates. What really concerns us in this check is to know that there is 1-1 map between x values and y values on the batching dimension. So this why I use set to eliminate the duplicates as we want to impose an existential clause for the training's and target's rows

Reply:
For the second the same applies. However in the second case we cannot make use of indices since `y` is a dense array so mainly we assume that it will contain all the values of the rows materialised. For the `x` I decided to use `np.unique` in order to compare the sizes between them. But it could similarly be expressed as `len(set(x_row_idx))`

Reply:
For consistency / minimizing differences I'd say either use `set()` on `np.unique` for both (with a slight preference for the latter). 

Two more comments:
- The `ValueError` message is copy-pasted in both cases; either provide a different message (if it makes things more clear), or refactor it so it's raised from only one place.
- The initial condition checks `self.y` although it is not used below; what you really care for is `y_row_idx`. Can't you tell if this is a dense or sparse case from `y_row_idx` instead of `self.y`?

Reply:
I will fix that for consistency. `Value error` can be also be raised from one place. For the `condition_check` I saw it as an Occams's Razor. It is the most explanatory way to indicate the difference. I cannot see a profound way to tell the difference using only the `y_row_idx` argument, which in both cases is an `nd.array` in the former indicating indices and the latter values.